### ISSUE TITLE
buildHelmChart: `namespace` not actually optional

### ISSUE THREAD

**Issue by Goorzhel** [farcaller/nix-kube-generators#3]
`namespace` is allowed to be null:

https://github.com/farcaller/nix-kube-generators/blob/6b7b51b332ff1eb023574c54981a67b6bb261b1a/lib/default.nix#L86

but is also [used in the derivation's own name](https://github.com/farcaller/nix-kube-generators/blob/6b7b51b332ff1eb023574c54981a67b6bb261b1a/lib/default.nix#L93), where a null value results in:
```
error:
       … while calling the 'derivationStrict' builtin

         at //builtin/derivation.nix:9:12: (source not available)

       … while evaluating the derivation attribute 'name'

         at /nix/store/y295j2wgd271jxqmman497xbcyr99pw9-source/pkgs/stdenv/generic/make-derivation.nix:348:7:

          347|     // (optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          348|       name =
             |       ^
          349|         let

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: cannot coerce null to a string
```

My first instinct was to send this patch:
```diff
diff --git a/lib/default.nix b/lib/default.nix
index daaf8c0..ec7cfff 100644
--- a/lib/default.nix
+++ b/lib/default.nix
@@ -90,7 +90,7 @@ rec {
     , apiVersions ? []
     }:
     pkgs.stdenv.mkDerivation {
-      name = "helm-${chart}-${namespace}-${name}";
+      name = if (!builtins.isNull namespace) then "helm-${chart}-${namespace}-${name}" else "helm-${chart}-${name}";

       passAsFile = [ "helmValues" ];
       helmValues = builtins.toJSON values;
```
but it seemed a bit ham-handed, and I'm not yet at the level of writing idiomatic Nix.

**Comment by farcaller** [comment]
Good catch! I think the patch makes sense, too. You can’t use the `or` operator in here as we want less data if it's NULL, not more.

I guess extracting it into a `let ... in` would improve the readability, though. I’m happy to accept such a PR.

**Comment by Goorzhel** [comment]
I didn't even think of a let-in -- thanks for the jump-start.

**Comment by farcaller** [comment]
Fixed in bba2e70a98ea6693cf755da50c8bb3e3bf1730a0
